lms/djangoapps/django_comment_client/utils.py

```diff
@@ -707,23 +714,27 @@ def prepare_content(content, course_key, is_staff=False, course_is_cohorted=None
         else:
             del endorsement["user_id"]
 
-    if course_is_cohorted is None:
-        course_is_cohorted = is_course_cohorted(course_key)
+    if discussion_division_enabled is None:
+        discussion_division_enabled = course_discussion_division_enabled(get_course_discussion_settings(course_key))
 
     for child_content_key in ["children", "endorsed_responses", "non_endorsed_responses"]:
         if child_content_key in content:
             children = [
-                prepare_content(child, course_key, is_staff, course_is_cohorted=course_is_cohorted)
+                prepare_content(child, course_key, is_staff, discussion_division_enabled=discussion_division_enabled)
                 for child in content[child_content_key]
             ]
             content[child_content_key] = children
 
-    if course_is_cohorted:
+    if discussion_division_enabled:
         # Augment the specified thread info to include the group name if a group id is present.
         if content.get('group_id') is not None:
-            content['group_name'] = get_cohort_by_id(course_key, content.get('group_id')).name
+            course_discussion_settings = get_course_discussion_settings(course_key)
+            content['group_name'] = get_group_name(content.get('group_id'), course_discussion_settings)
+            content['is_commentable_divided'] = is_commentable_divided(
```

Comment:
Note to reviewers: here is where the bug is fixed (referenced above with regard to tests).